### PR TITLE
More fixes for labelmaps

### PR DESCRIPTION
**Pull request by scottwittenburg**
This PR:

- [x] handles the labelmap opacity slider the way we want
- [x] debounces most of the application sliders for a better user experience
- [x] fixes the z-index of the categorical colormap selection component
- [x] fixes tooltip positions of some of the top row buttons so they can be seen

### CODE REVIEW

**Comment by scottwittenburg** [comment]
@thewtex I still don't think I understand why the volume rendering looks the way it does when we have a label map, something still seems weird (to me) there.

Also, this PR would require a [change](https://github.com/Kitware/vtk-js/pull/1389) to `vtk.js` unless you'd rather I just supply that pwf from the application.

Otherwise I think this addresses most of the things we discussed today.  Let me know if you see something I may have missed.

**Comment by scottwittenburg** [comment]
Hopefully this shows off the labelmap opacity slider working the way you want, as well as highlighting how debouncing the slider actions makes them more responsive.

localhost_8080

**Comment by scottwittenburg** [comment]
@thewtex Just to be clear, this shouldn't be merged until we 1) either accept that vtk.js PR or create the pwf in the application, and 2) rewrite the history of this PR branch so it's clean and we get a new version.

File: src/UserInterface/Image/createGradientOpacitySlider.js [hunk 1]
```diff
@@ -27,9 +28,9 @@ function createGradientOpacitySlider(store, uiContainer) {
     store.imageUI.representationProxy.setEdgeGradient(gradientOpacity)
     store.renderWindow.render()
   }
-  autorun(() => {
-    updateGradientOpacity()
-  })
+  reaction(() => {
+    return store.imageUI.gradientOpacity
+  }, macro.debounce(updateGradientOpacity, 25, false))
```
Comment (thewtex) [inline src/UserInterface/Image/createGradientOpacitySlider.js:33]: Could this be causing the lag we were observing the gradient opacity slider or this helping to address it?

File: src/UserInterface/Image/createLabelMapColorWidget.js [hunk 2]
```diff
@@ -13,6 +14,12 @@ function createLabelMapColorWidget(store, uiContainer) {
 
   const labelMapColorUIGroup = document.createElement('div')
   labelMapColorUIGroup.setAttribute('class', style.uiGroup)
+  // This group has a sub-component that drops down and needs to appear
+  // above components in the "uiGroup" below.
+  labelMapColorUIGroup.setAttribute(
+    'style',
+    'position: relative; z-index: 1001;'
+  )
```
Comment (thewtex) [inline src/UserInterface/Image/createLabelMapColorWidget.js:22]: :+1:

File: src/UserInterface/Image/createLabelMapColorWidget.js [hunk 3]
```diff
@@ -101,16 +113,23 @@ function createLabelMapColorWidget(store, uiContainer) {
   applyContrastSensitiveStyle(store, 'invertibleButton', sliderEntryDiv)
   const volume = store.imageUI.representationProxy.getVolumes()[0]
   const volumeProperty = volume.getProperty()
+  const sliceActor = store.imageUI.representationProxy.getActors()[0]
+  const sliceProperty = sliceActor.getProperty()
   function updateLabelMapColorOpacity() {
     const labelMapOpacity = store.imageUI.labelMapOpacity
     opacityElement.value = labelMapOpacity
     const numberOfComponents = store.imageUI.numberOfComponents
+    for (let c = 0; c < numberOfComponents; ++c) {
```
Comment (thewtex) [inline src/UserInterface/Image/createLabelMapColorWidget.js:122]: Nitpick: In JavaScript `c++` is preferred for performance, the opposite of C++ :-(

File: src/createViewer.js [hunk 4]
```diff
@@ -97,6 +98,8 @@ const createViewer = (
               (numberOfComponents === 3 || numberOfComponents === 4)
             ) {
               preset = 'Grayscale'
+            } else if (numberOfComponents === 1 && !!store.imageUI.labelMap) {
+              preset = 'Grayscale'
```
Comment (thewtex) [inline src/createViewer.js:102]: :+1:

File: src/createViewer.js [hunk 5]
```diff
@@ -137,14 +140,14 @@ const createViewer = (
...
             volumeProperty.setComponentWeight(component, visibility)
-            //volumeProperty.setIndependentComponents(true);
           }
 
           // Now for the slice rendering
           updateSliceProperties(store)
         }
 
         if (!!store.imageUI.labelMap) {
+          console.log(`we have a label map, num comps = ${numberOfComponents}`)
```
Comment (thewtex) [inline src/createViewer.js:150]: debug code

File: src/UserInterface/Image/createGradientOpacitySlider.js (hunk 1 above)
Comment (scottwittenburg) [reply src/UserInterface/Image/createGradientOpacitySlider.js:33]: For me it was great improved by adding this debounce.  In fact, all the sliders were really laggy until I debounced them in a similar fashion.

**Comment by scottwittenburg** [comment]
@thewtex This is either ready for another round of review, or just merge it and the next things can happen in a new topic.

**Pull request merged**